### PR TITLE
Added mouseover event for page titles to show day name

### PR DESCRIPTION
**Pull request by braydie**
Introduces a mouseover event that will parse page refs and titles for date and insert the day name as the title to that element

image
image

### CODE REVIEW

File: src/ts/core/features/day-title.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import {RoamDate} from '../roam/date'
+
+const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
+
+function getDayFromDate(name: string) {
+    let re = /(.*) (\d+).{2}, (\d{4})/i
```
[the remark this answers is not in the document]
Comment (braydie) [reply src/ts/core/features/day-title.ts:6]: Yeah it's looking for an exact match here to filter out non-date page titles. Prior to integrating with Roam Toolkit I was using the matches from the regex to create a Date object. Can you think of a better way of filtering out non-date pages?

File: src/ts/core/features/day-title.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        const d = RoamDate.parse(name)
+        return days[d.getDay()]
+    }
+    return null
+}
+
+function isPageViewTitle(element: HTMLElement) {
+    return (
+        element.parentNode &&
+        element.parentNode.parentNode &&
```
[the remark this answers is not in the document]
Comment (braydie) [reply src/ts/core/features/day-title.ts:18]: Nice, I forgot I could use it!

File: src/ts/core/features/day-title.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import {RoamDate} from '../roam/date'
+
+const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
```
[the remark this answers is not in the document]
Comment (Stvad) [reply src/ts/core/features/day-title.ts]: you moved it to a different file :p

[the remark this answers is not in the document]
Comment (braydie) [reply src/ts/core/features/day-title.ts]: woops! gimme a minute...

File: src/ts/core/features/day-title.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Stvad) [reply src/ts/core/features/day-title.ts:24]: current way is fine. one other option would be to re-use fuzzydate functionality and try parsing any string we get, but probably doesn't really make sense in this case

**Comment by Stvad** [comment]
Looks good to me now! Thank you again!

File: src/ts/core/features/day-title.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (braydie) [reply src/ts/core/features/day-title.ts:24]: Happy to leave it as is if you are ✌

**Comment by Stvad** [comment]
initiated publishing to the Chrome store - that takes a bit https://chrome.google.com/webstore/detail/ebckolanhdjilblnkcgcgifaikppnhba/publish-delayed?authuser=2&hl=en

**Pull request merged**